### PR TITLE
fix: tablefacet销毁时off事件

### PR DESCRIPTION
**Pull request by serializedowen**
### 👀 PR includes

✨ Feature

- [ ] New feature

🎨 Enhance

- [ ] Code style optimization
- [ ] Refactoring
- [ ] Change the UI
- [ ] Improve the performance
- [ ] Type optimization

🐛 Bugfix

- [ ] Solve the issue and close #0

🔧 Chore

- [ ] Test case
- [ ] Docs / demos update
- [ ] CI / workflow
- [ ] Release version
- [ ] Other ()

### 🖼️ Screenshot

| Before | After |
| ------ | ----- |
| ❌      | ✅     |

### 🔍 Self-Check before the merge

- [ ] Add or update relevant docs.
- [ ] Add or update relevant demos.
- [ ] Add or update test case.
- [ ] Add or update relevant TypeScript definitions.

### CODE REVIEW

File: packages/s2-core/src/facet/table-facet.ts [hunk 1]
```diff
@@ -86,87 +87,95 @@ export class TableFacet extends BaseFacet {
...
+      (s2.dataSet as TableDataSet).getDisplayDataSet(),
+    );
+  };
 
   get dataCellTheme() {
     return this.spreadsheet.theme.dataCell.cell;
   }
 
+  public destroy(): void {
+    const s2 = this.spreadsheet;
```
Comment (xingwanying) [inline packages/s2-core/src/facet/table-facet.ts:173]: https://github.com/antvis/S2/pull/1678
这个问题会不会又出现

Comment (serializedowen) [reply packages/s2-core/src/facet/table-facet.ts:173]: 不会 orz

File: packages/s2-core/src/facet/table-facet.ts [hunk 2]
```diff
@@ -86,87 +87,95 @@ export class TableFacet extends BaseFacet {
...
-      set(s2.dataCfg, 'sortParams', [...oldConfigs, ...params]);
-      s2.setDataCfg(s2.dataCfg);
-      s2.render(true);
-      s2.emit(
-        S2Event.RANGE_SORTED,
-        (s2.dataSet as TableDataSet).getDisplayDataSet(),
-      );
+    const oldConfigs = currentParams.filter((config) => {
+      const newItem = params.find((p) => p.sortFieldId === config.sortFieldId);
+      if (newItem) {
```
Comment (stone-lyl) [inline packages/s2-core/src/facet/table-facet.ts:121]: return !!newItem

File: packages/s2-core/src/facet/table-facet.ts [hunk 3]
```diff
@@ -86,87 +87,95 @@ export class TableFacet extends BaseFacet {
...
-          // remove filter params on current key if passed an empty filterValues field
-          oldConfig.splice(oldIndex);
-        } else {
-          // if filter with same key already exists, replace it
-          oldConfig[oldIndex] = params;
-        }
+    if (oldIndex !== -1) {
+      if (unFilter) {
+        // remove filter params on current key if passed an empty filterValues field
+        oldConfig.splice(oldIndex);
```
Comment (stone-lyl) [inline packages/s2-core/src/facet/table-facet.ts:150]: 这里和注释好像不一致。
注释表达的删除当前的 filter params ,但是 `oldConfig.splice(oldIndex)` 会删除当前 index 当最后的 `oldConfig` 吧？

Comment (serializedowen) [reply packages/s2-core/src/facet/table-facet.ts:150]: 这里代码没有改 0 0就是移动了下位置

Comment (stone-lyl) [reply packages/s2-core/src/facet/table-facet.ts:150]: 你要不要加一个单测啊

**Review by stone-lyl — APPROVED** [review]

File: packages/s2-core/src/facet/table-facet.ts (hunk 3 above)
Comment (serializedowen) [reply packages/s2-core/src/facet/table-facet.ts:150]: 已经加了

**Pull request merged**